spark/src/main/scala/ai/chronon/spark/Join.scala

```diff
@@ -212,9 +212,19 @@ class Join(joinConf: api.Join,
       val wheres = Seq(s"ds >= '${effectiveRange.start}'", s"ds <= '${effectiveRange.end}'")
       val sql = QueryUtils.build(null, partTable, wheres)
       logger.info(s"Pulling data from joinPart table with: $sql")
-      val df = tableUtils.sparkSession.sql(sql)
-      (joinPart, df)
-    }
+      val dfTry = Try {
+        tableUtils.sparkSession.sql(sql)
+      }
+
+      dfTry match {
+        case Success(df) => Some((joinPart, df))
+        case Failure(e: org.apache.spark.sql.AnalysisException) if e.getMessage.contains("Table or view not found") =>
```

Comment:
instead of relying on string matching, can we explicitly check existence by using `TableUtils.tableExists`?

Reply:
Pushed the change